rfd/9999-oidc-idp-for-aws-rds-dbs-discovery.md

```diff
@@ -0,0 +1,495 @@
+}
+```
+
+- Issuer: the provider's URL
+- JWKS URI: the endpoint where the provider returns the public keys.
+- Claims Supported: a list of supported claims that the OpenID Provider (ie, Teleport) MAY be able to supply values for when issuing a token:
+  - `iss`: the issuer identity, on our case it will be the same as the `issuer` key.
+  - `sub`: identifies the subject of the token, on our case it will contain the user's name.
+  - `aud`: the audience for the token, on our case it will be the same as the `audience` defined when configuring the OIDC IdP in AWS (step 3.3 for the User's Point of View).
+  - `jti`: is the token id.
+  - `iat`: unix epoch time of when the token was issued.
+  - `exp`: unix epoch time of when the token is no longer valid (expired).
+  - `nbf`: unix epoch time which indicates when the token began to be valid.
+- ID Token Signing Algorithm Values Supported: a list of supported signing algorithms to assist during token validation.
+- Scopes Supported: a list of scopes to be used, on our case it will only contain the required one: `openid`.
+- Subject Types Supported: a list of supported subject types:
+  - `public`: we can either return clear text version of the subject (user names) or
+  - `pairwise`: we can return a hash of the subject's identity (to hide their email, for instance)
+
+##### JSON Web Key Sets
+This endpoint returns a list of public keys (usually only one key, except for periods of key rotation).
+
+This endpoint URI must be equal to `jwks_uri` defined in the previous section.
+
+As an example, `<teleportProviderURL>/.well-known/jwks-oidc`.
+
+It should return the following:
+
+```
+200 OK
+
+{
+  "keys": [
+    {
+      "kty": "RSA",
+      "alg": "RS256",
+      "n": "<public key value>",
+      "e": "AQAB"
+    }
+  ]
+}
+```
+
+This must respect the RFC7517 (eg of a key can be found at https://www.rfc-editor.org/rfc/rfc7517#appendix-A.1).
+
+Obtaining this key should be possible using
+```golang
+GetCertAuthority(
+	ctx,
+	types.CertAuthID{
+		Type:       types.OIDCSigner, // new key type
+		DomainName: clusterName,
+	},
+	false /*loadKeys*/,
+)
+```
+
+#### AWS Role for Teleport OIDC IdP
+
+The user will create or associate an AWS Role to the Teleport OIDC IdP.
+This role must have access to list RDS Databases and have a policy that trusts Teleport as an Identity Provider.
+
+At least a policy allowing the following must be part of the role:
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": "rds:DescribeDBInstances",
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+The AWS Role must trust Teleport as an IdP.
+To do so, the user must add the following Trusted relationship:
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn:aws:iam::<account id>:oidc-provider/<teleportProviderURL>"
```

Comment:
Do you embed the actual proxy address here and below? Can you use specific values for better demonstration? E.g. say we have a proxy at `https://proxy.example.com:3080`.